### ISSUE TITLE
[iOS/Android] Missing functionalities

### ISSUE THREAD

**Issue by umang-simform** [notifee/react-native-notifee#29]
We are using notifee in our production app and even being in alpha it is serving its purpose very well.
As we all know there are many things still pending but I think right now it misses 2 basic yet important features.

1. cancelAllNotifications function is not working on iOS (we want to clear all notifications when app is opened)
2. ~~Not able to get the current badge count (we want to perform some activity when user have badge count while launching the app)~~
3. ~~Ability to set badge count (It was supported in RNFirebase) but in the current version of notifee we are not able to set badge.~~

We are using 0.4.0 and I can say we are able to receive/display a notification in android/iOS perfectly for foreground as well as background.
We are looking forward to hearing from the dev team as according to us these are core features that will be required by most of the apps.

**Comment by johanschram** [comment]
Getting / setting  badge count would be indeed nice :)

**Comment by Ehesp** [comment]
We're planning to implementing badge count support on iOS, however not on Android as it's non-standard. Our plan was to do something like:

```
notifee.setBadgeCount(count: number): Promise<void>; // 0 to remove
notifee.getBadgeCount(): Promise<number>;
notifee.incrementBadgeCount(): Promise<void>; // increases by one via native
notifee.decrementBadgeCount(): Promise<void>; // decreases by one via native
```

**Comment by Salakar** [comment]
#32 was just merged - adds support for iOS - we'll have a release out within the next day.

Still not sure if this is viable for Android as it's a non-standard API and requires a third party library like ShortcutBadger - do people actually need/use this for Android?

**Comment by mikehardy** [comment]
From my take on messaging traffic and how delighted my colleagues are that use Android phones that have launchers with badging, it's a desire but the support is terrible and it bugs me. Basically a user plus vs an engineer negative. Best effort with a big "no complaining it sucks because it sucks not because the library sucks" disclaimer? I cringe typing that.

**Comment by johanschram** [comment]
> #32 was just merged - adds support for iOS - we'll have a release out within the next day.
> 
> Still not sure if this is viable for Android as it's a non-standard API and requires a third party library like ShortcutBadger - do people actually need/use this for Android?

As an IOS user, I would definitly feel like something is missing if it wasn't there. As an android user, I would still be trying to deal with the batery saver / security maze menu / whatever else is stopping me to properly get proper background notifications... So, badge count would be nice, but well, the devil is in the details of non-standard...

**Comment by umang-simform** [comment]
@Salakar  I am glad that you have looked into it but It is not all, We need to clear all notifications when app launches and it is really important for us.

**Comment by Ehesp** [comment]
> From my take on user support traffic and how delighted my colleagues are that use Android phones that have launchers with badging, it's a desire but the support is terrible and it bugs me. Basically a user plus vs an engineer negative. Best effort with a big "no complaining it sucks because it sucks not because the library sucks" disclaimer? I cringe typing that.

Yep and the other aspect of it is that ShortcutBadger hasn't been updated in over a year. Not worth it imo.

**Comment by Krunal-Kevadiya** [comment]
@Ehesp What about the first point? (cancelAllNotifications function is not working(Unavailable) on iOS [we want to clear all notifications when the app is opened])

**Comment by Salakar** [comment]
@Krunal-Kevadiya working on a bunch of FCM & Notifee stuff today so will take a look at that also - hoping to have a new release for RNFirebase Messaging & Notifee out later today with a bunch of fixes

RNFB work for reference: https://github.com/invertase/react-native-firebase/pull/3427

**Comment by Krunal-Kevadiya** [comment]
@Salakar: Thanks, sure I am looking at that.

**Comment by Salakar** [comment]
iOS Badge functionality has been released in 0.5.0 - thanks

**Comment by Salakar** [comment]
@umang-simform cancelAllNotifications fixed for iOS in 0.5.1 release - thanks
